README.md

```diff
@@ -796,6 +796,122 @@ The `profunctor` method takes two arguments:
 
 3. `promap` must return a value of the same Profunctor
 
+## Types
+
+In Fantasy Land, types are specified by an object containing a catamorphic
+method `cata`. `cata` has an arity matching the number of constructor functions
+belonging to each type. Each argument to `cata` is either:
```

Comment:
For the benefit of people who don't know the difference between Types and Type Classes, I think we should elaborate the difference here to avoid confusion.  Maybe a paragraph explaining the benefits in having a specification for both?

Reply:
I had considered placing a type specification in a separate file but figure that could be a separate PR.

I agree that an introductory paragraph contrasting the specification of algebras with that of types would be useful. Suggestions are welcome 😄